modules/javafx.base/src/main/java/javafx/beans/value/ObservableValue.java

```diff
@@ -137,4 +139,105 @@
+     * given mapping function on this value. The result is updated when this
+     * {@code ObservableValue} changes. If this value is {@code null}, no
+     * mapping is applied and the resulting value is also {@code null}.
+     * <p>
+     * For example, mapping a string to an upper case string:
+     * <pre>{@code
+     * var text = new SimpleStringProperty("abcd");
+     * ObservableValue<String> upperCase = text.map(String::toUpperCase);
+     *
+     * upperCase.getValue();  // Returns "ABCD"
+     * text.set("xyz");
+     * upperCase.getValue();  // Returns "XYZ"
+     * text.set(null);
+     * upperCase.getValue();  // Returns null
+     * }</pre>
+     *
+     * @param <U> the type of values held by the resulting {@code ObservableValue}
+     * @param mapper the mapping function to apply to a value, cannot be {@code null}
+     * @return an {@code ObservableValue} that holds the result of applying the given
+     *     mapping function on this value, or {@code null} when it
+     *     is {@code null}; never returns {@code null}
+     * @throws NullPointerException if the mapping function is {@code null}
+     */
+    default <U> ObservableValue<U> map(Function<? super T, ? extends U> mapper) {
+        return new MappedBinding<>(this, mapper);
+    }
+
+    /**
+     * Returns an {@code ObservableValue} that holds this value, or the given constant if
+     * it is {@code null}. The result is updated when this {@code ObservableValue} changes. This
+     * method, when combined with {@link #map(Function)}, allows handling of all values
+     * including {@code null} values.
+     * <p>
+     * For example, mapping a string to an upper case string, but leaving it blank
+     * if the input is {@code null}:
+     * <pre>{@code
+     * var text = new SimpleStringProperty("abcd");
+     * ObservableValue<String> upperCase = text.map(String::toUpperCase).orElse("");
+     *
+     * upperCase.getValue();  // Returns "ABCD"
+     * text.set(null);
+     * upperCase.getValue();  // Returns ""
+     * }</pre>
+     *
+     * @param constant the value to use when this {@code ObservableValue}
+     *     holds {@code null}; can be {@code null}
+     * @return an {@code ObservableValue} that holds this value, or the given constant if
+     *     it is {@code null}; never returns {@code null}
+     */
+    default ObservableValue<T> orElse(T constant) {
+        return new OrElseBinding<>(this, constant);
+    }
+
+    /**
+     * Returns an {@code ObservableValue} that holds the value of an {@code ObservableValue}
+     * produced by applying the given mapping function on this value. The result is updated
+     * when either this {@code ObservableValue} or the {@code ObservableValue} produced by
+     * the mapping changes. If this value is {@code null}, no mapping is applied and the
+     * resulting value is {@code null}. If the mapping resulted in {@code null}, then the
+     * resulting value is also {@code null}.
+     * <p>
```

Comment:
It might be worth borrowing some language from `Optional::flatMap`. Maybe something like this?

```
This method is similar to {@link #map(Function)}, but the mapping function is
one whose result is already an ObservableValue, and if invoked, flatMap does
not wrap it within an additional ObservableValue.
```

Reply:
Added this and put code tags where necessary.